lib/backend/k8s/syncer.go

```diff
@@ -477,264 +507,322 @@ func (syn *kubeSyncer) readFromKubernetesAPI() {
 			// Event is OK - parse it and send it over the channel.
 			kvp = syn.parseNetworkPolicyEvent(event)
 			latestVersions.networkPolicyVersion = kvp.Revision.(string)
-			syn.sendUpdates([]model.KVPair{*kvp})
+			syn.sendUpdates([]model.KVPair{*kvp}, KEY_NP)
 		case event = <-snpChan:
 			log.Debugf("Incoming SystemNetworkPolicy watch event. Type=%s", event.Type)
 			if syn.eventNeedsResync(event) {
-				needsResync = true
+				syn.needsResync[KEY_SNP] = true
 				continue
 			} else if syn.eventRestartsWatch(event, KEY_SNP) {
+				// Resources backed by TPRs need to be resynced on empty events.
+				syn.needsResync[KEY_SNP] = true
 				syn.closeWatcher(KEY_SNP)
 				continue
 			}
 			// Event is OK - parse it and send it over the channel.
 			kvp = syn.parseSystemNetworkPolicyEvent(event)
 			latestVersions.systemNetworkPolicyVersion = kvp.Revision.(string)
-			syn.sendUpdates([]model.KVPair{*kvp})
+			syn.sendUpdates([]model.KVPair{*kvp}, KEY_SNP)
 		case event = <-gcChan:
 			log.Debugf("Incoming GlobalConfig watch event. Type=%s", event.Type)
 			if syn.eventNeedsResync(event) {
-				needsResync = true
+				syn.needsResync[KEY_GC] = true
 				continue
 			} else if syn.eventRestartsWatch(event, KEY_GC) {
+				// Resources backed by TPRs need to be resynced on empty events.
+				syn.needsResync[KEY_GC] = true
 				syn.closeWatcher(KEY_GC)
 				continue
 			}
 			// Event is OK - parse it and send it over the channel.
 			kvp = syn.parseGlobalConfigEvent(event)
 			latestVersions.globalConfigVersion = kvp.Revision.(string)
-			syn.sendUpdates([]model.KVPair{*kvp})
+			syn.sendUpdates([]model.KVPair{*kvp}, KEY_GC)
 		case event = <-poolChan:
 			log.Debugf("Incoming IPPool watch event. Type=%s", event.Type)
 			if syn.eventNeedsResync(event) {
-				needsResync = true
+				syn.needsResync[KEY_IP] = true
 				continue
 			} else if syn.eventRestartsWatch(event, KEY_IP) {
+				// Resources backed by TPRs need to be resynced on empty events.
+				syn.needsResync[KEY_IP] = true
 				syn.closeWatcher(KEY_IP)
 				continue
 			}
 			// Event is OK - parse it and send it over the channel.
 			kvp = syn.parseIPPoolEvent(event)
 			latestVersions.poolVersion = kvp.Revision.(string)
-			syn.sendUpdates([]model.KVPair{*kvp})
+			syn.sendUpdates([]model.KVPair{*kvp}, KEY_IP)
 		case event = <-noChan:
 			log.Debugf("Incoming Node watch event. Type=%s", event.Type)
 			if syn.eventNeedsResync(event) {
-				needsResync = true
+				syn.needsResync[KEY_NO] = true
 				continue
 			} else if syn.eventRestartsWatch(event, KEY_NO) {
+				syn.needsResync[KEY_NO] = true
```

Comment:
Oh huh, good spot.  We probably don't need to do that (though I don't think it hurts).